src/main/java/io/openliberty/tools/intellij/util/LibertyProjectUtil.java

```diff
@@ -149,6 +153,27 @@ public static ShellTerminalWidget getTerminalWidget(Project project, LibertyModu
         TerminalView terminalView = TerminalView.getInstance(project);
         // look for existing terminal tab
         ShellTerminalWidget widget = getTerminalWidget(libertyModule, terminalView);
+        TerminalToolWindowManager manager = TerminalToolWindowManager.getInstance(project);
+        ToolWindow toolWindow = manager.getToolWindow();
+
+        if (toolWindow != null) {
+            ContentManager contentManager = toolWindow.getContentManager();
+            Content[] contents = contentManager.getContents();
+
+            int index = 0;
+            for (int i = 0; i < contents.length; i++) {
+                if (contents[i].getPreferredFocusableComponent().equals(widget)) {
+                    index = i;
+                    break;
+                }
+            }
+            if (contents.length > 0 && index > 0) {
+                Content terminalContent = contents[index];
+                contentManager.setSelectedContent(terminalContent);
+                terminalContent.getComponent().requestFocus();
+            }
+        }
+
```

Comment:
It would be better to separate the set focus code to its own method in `LibertyProjectUtil.java` which can be called from `getTerminalWidget()` in `LibertyGeneralAction.java`. Can perhaps try using `widget.hasFocus()` to check if the terminal already has focus before calling `setFocus()`.

Reply:
Should also rename the `getTerminalWidget()` method in `LibertyGeneralAction.java` to reflect the fact that it is setting focus in addition to retrieving the terminal widget.

Reply:
`widget.hasFocus()` always returns false, even though the corresponding terminal widget is `visible` and focused.

Reply:
I see that you moved the code into `setFocusToWidget()` but its still being called directly by `getTerminalWidget()`. This meant to be moved up to the method you renamed in `LibertyGeneralAction.java`. Did you try this out? Was there an issue with doing this?

Reply:
I made the changes as per your suggestions